lib/identity/client.js

```diff
@@ -602,11 +606,83 @@ class Client extends PartialClient {
       currentGroup.groupID,
       recordType
     )
-    if (secretType === 'File') {
-      await this.storage.share(recordType, clientID)
-    }
     return sharedWithGroup.record_type
   }
+  /**
+   * addSecretToNamespace takes a user defined namespace name and a secret to add to the namespace
+   *
+   * @param {String} namespace The user given name for the secret stored in the meta data
+   * @param {String} secretType The secret type, example 'Credential'
+   * @param {String} secretName The user given name for the secret stored in the meta data
+   *
+   * @return {Boolean}
+   */
+  async addSecretToNamespace(secretName, secretType, namespace) {
+    // Check if the namespace exists
+    let currentGroup = await this.getOrCreateNamespace(namespace)
+    // Generate the Record Type to share
+    const recordType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+    let sharedWithGroup = await this.storage.shareRecordWithGroup(
+      currentGroup.groupID,
+      recordType
+    )
+    return sharedWithGroup.record_type
+  }
+  /**
+   * addIdentityToNamespace takes a user defined namespace name and a username for an identity and gives access to the given namespace for the identity
+   *
+   * @param {String} namespace The user given name for the secret stored in the meta data
+   * @param {String} usernameToAdd The username of the user that will have permissions added
+   *
+   * @return {Boolean}
+   */
+  async addIdentityToNamespace(usernameToAdd, namespace) {
+    // Validation
+    if (usernameToAdd == '') {
+      throw new Error('Username Required')
+    }
+    // Look Up the ClientID for Username
+    let clientResponse = await this.searchRealmIdentitiesByUsername([
+      usernameToAdd,
+    ])
+    let usernameLowerCase = usernameToAdd.toLowerCase()
+    let clientID
+    if (clientResponse.searched_identities_information === null) {
```

Comment:
ideally the response type should handle setting some reasonable default value like the empty array  to avoid having to write this validation logic in our or customers code that uses the sdk